### DISCUSSION TITLE
Instructions unclear

### DISCUSSION THREAD

**Discussion by dreenot** [alchemyguy/YoutubeLaravelApi#26] [General]
Hello, I'm interested in using your package. It seems very useful. But I didn't understand the instructions for using the package

_"Then add these values - client_id, client_secret, api_key and redirect_url in the env file and you can start using the package now."_

The client_id, client_secret, api_key and redirect_url comes from the API dashboard on Google's site. And the .env is a file of Laravel's installation. But what am I supposed to do with it? How should I add it? I can't just type it there, I must make it into a constant or something. How is the constant supposed to be called? I can't just give it a random name.

_# Replace the identifier with a unqiue identifier for account or channel
$authUrl = $authObject->getLoginUrl('email','identifier');_ 

What identifier and email are we talking about? Is it one of the things from my API like client_id and client_secret or can I type anything? Is it a data from the client, like their channel id? About the email, is it my email, or the user's email? Does it need to be the email that created the Google API? Can it be one of the others set as owner for the Google API?

**Comment by alchemyguy** [comment]
Hi @dreenot ,

> The client_id, client_secret, api_key and redirect_url comes from the API dashboard on Google's site. And the .env is a file of Laravel's installation. But what am I supposed to do with it? How should I add it? I can't just type it there, I must make it into a constant or something. How is the constant supposed to be called? I can't just give it a random name.

yes, you have to just add them in .env and package will use it from there.

>What identifier and email are we talking about? Is it one of the things from my API like client_id and client_secret or can I type anything? Is it a data from the client, like their channel id? About the email, is it my email, or the user's email? Does it need to be the email that created the Google API? Can it be one of the others set as owner for the Google API?

 `email` is `client email` which you want to authorise.  
` identifier` can be any unique id for the user, that you will get along with auth code in redirect_url ( you can use it to identify which user authorised the account ).

>$authUrl = $authObject->getLoginUrl('email','identifier');
when user visits this auth url and authorises permissions, auth code is send to redirect url along with identifier on authorisation.

**Reply by dreenot** [reply]
Hi @alchemyguy.
Thank you, your package is amazing.

The only problem I am having now is with pagination. I can only get 50 results from Google Api, and to access more I have to send the NextPageToken as a parameter. But there is not place for them on the parameters, so I can't loop it to get more pages, making me only get the first page.
I'll try to workaround it myself, if it works well I'll try to send a pull request.

There is a tiny mistake on read me tho. The parameters on **"Channel Subscription List List of subscriptions of the channel"**
are 'channelId' and 'maxResults' instead of 'channel_id' and 'max_results'
